geo/src/main/java/org/apache/carbondata/geo/GeoHashUtils.java

```diff
@@ -83,6 +85,17 @@ public static long lonLat2GeoID(long longitude, long latitude, double oriLatitud
     return colRow2GeoID(ij[0], ij[1]);
   }
 
+  public static void validateUDFInputValue(Object input, String inputName, String datatype)
+      throws MalformedCarbonCommandException {
+    if (inputName.equalsIgnoreCase("gridSize") && (input == null
+        || Integer.parseInt(input.toString()) <= 0)) {
```

Comment:
can we use Pattern.compile(POSITIVE_INTEGER_REGEX) to validate here

Reply:
Done